pkg/boot/kexec/memory_linux.go

```diff
@@ -667,3 +667,52 @@ func (m *MemoryMap) Insert(r TypedRange) {
 	newMap.sort()
 	*m = newMap
 }
+
+// Memory type (uint32) in UefiPayload
+const (
+	PayloadTypeRAM      = 1
+	PayloadTypeDefault  = 2
+	PayloadTypeAcpi     = 3
+	PayloadTypeNVS      = 4
+	PayloadTypeReserved = 5
+)
+
+// payloadMemoryMapEntry represent a memory map entry in payload param
+type payloadMemoryMapEntry struct {
```

Comment:
Could this use the `TypedRange` above?

Reply:
See the above comments, I agreed that we should merge them. I feel it's weird to include them in this PR.